packages/@react-aria/button/src/useButton.ts

```diff
@@ -93,8 +93,26 @@ export function useButton(props: AriaButtonProps<ElementType>, ref: RefObject<an
   });
 
   let {focusableProps} = useFocusable(props, ref);
-  let buttonProps = mergeProps(focusableProps, pressProps);
-  buttonProps = mergeProps(buttonProps, filterDOMProps(props, {labelable: true}));
+  // enter triggers form submit on key down
+  // since enter is a special key for submitting forms and it occurs on key down
+  // spacebar trigger form submit on key up, this is because it's actually a 'click'
+  let submitProps = {onKeyDown: undefined, onKeyUp: undefined};
+  if (type === 'submit') {
```

Comment:
can you use this https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/interactions/src/usePress.ts#L292
and check if it is a submit in usePress?
(from devon's head)

Reply:
Not sure I follow, that's in pointerdown, this is in keydown/keyup

I don't think there's a problem with clicking on the submit button non-keyboard, at least, i haven't found one yet